### ISSUE TITLE
Implement runtime flags

### ISSUE THREAD

**Issue by mwobensmith** [mozilla/iris2#94]
This will take many check-ins, but here's the list.

Core:

```
    parser.add_argument('-a', '--rerun',
                        help='Rerun last failed tests',
                        action='store_true')
    parser.add_argument('-b', '--highlight',
                        help='Highlight patterns and click actions',
                        action='store_true')
    parser.add_argument('-c', '--clear',
                        help='Clear run data',
                        default=False,
                        action='store_true')
    parser.add_argument('-e', '--email',
                        help='Submit email report',
                        action='store_true')
    parser.add_argument('-g', '--image_debug',
                        help='Temporary flag to hunt down misplaced pattern images.',
                        action='store_true')
    parser.add_argument('-l', '--locale',
                        help='Locale to use for pattern search',
                        action='store',
                        default='en-US')
    parser.add_argument('-m', '--mouse',
                        help='Change mouse speed',
                        type=float,
                        action='store',
                        default=0.5)
    parser.add_argument('-o', '--override',
                        help='Override disabled tests',
                        action='store_true')
    parser.add_argument('-p', '--port',
                        help='Port to use for local web server',
                        type=int,
                        action='store',
                        default=2000)
    parser.add_argument('-r', '--report',
                        help='Report tests to TestRail',
                        action='store_true')
    parser.add_argument('-w', '--workdir',
                        help='Path to working directory',
                        type=os.path.abspath,
                        action='store',
                        default='%s/.iris2' % home)
    parser.add_argument('-x', '--exclude',
                        help='List of test names or directories to exclude',
                        metavar='empty')
    parser.add_argument('-z', '--resize',
                        help='Convert hi-res images to normal',
                        action='store_true')
```

From Firefox:
```
    parser.add_argument('-f', '--firefox',
                        help='Firefox version to test',
                        action='store',
                        default='latest-beta')
    parser.add_argument('-u', '--update_channel',
                        help='Update channel profile preference',
                        action='store')
```

**Comment by mwobensmith** [comment]
Hmmm... dunno what happened to `-t`, which is the most useful one right now. Otherwise we run the entire test suite. :\

@silviuFlorin can you start with `-t` and `-x`? Also, I am thinking that we might change the implementation in Iris 2.0 a little so that any arguments passed to these flags is used to match against any strings in the tests or test paths, not just full test names or directories.

Note: this match only applies to the test path after the tests directory itself. 

Sample paths of tests:
```
/Users/mwobensmith/PycharmProjects/iris2/tests/firefox/awesomebar/disable_one_off_searches_from_awesomebar.py
/Users/mwobensmith/PycharmProjects/iris2/tests/firefox/awesomebar/disable_web_search_awesomebar.py
```

Some examples using the above path:

`
iris -t awesomebar
`

This will select all tests with "awesomebar" in their path, which includes the test case name or any package that they exist in. This matches both tests above.

`Iris -x disable_web`

This will take all tests in the repo, and exclude using the string "disable_web." In the example above, it would select one test and exclude the other.

`Iris -t awesomebar -x one_off`

Select all tests with string "awesomebar" in their path and then remove anything that contains the string "one_off." This will exclude the first test above, but include the second.

`iris -t PycharmProjects`

This will select zero tests, as the string to be matched is outside of the target test folder and doesn't match any tests.

**Comment by silviuFlorin** [comment]
The handling logic for test and exclude arguments was merged into master

**Comment by mwobensmith** [comment]
Fixed in PR #419
